hm_pyhelper/util/sentry.py

```diff
@@ -0,0 +1,97 @@
+from typing import Dict, Union
+import hashlib
+
+EVENT_COUNT_KEY = "event_count"
+SUPPRESSION_COUNT_KEY = "suppression_count"
+
+# per session event counters, didn't feel the need to persist them
+event_counters = {}
+
+# maximum suppression interval. It grows exponentially till this value is reached.
+max_suppression_count = 128
+
+
+def sentry_fingerprint(hints: Dict) -> Union[str, None]:
+    """ return sha256 of log_record/exc_info as fingerprint"""
+    fingerprint_info_key = 'log_record'
+    if 'exc_info' in hints:
+        fingerprint_info_key = 'exc_info'
+
+    value_to_fingerprint = hints.get(fingerprint_info_key, None)
+
+    if not value_to_fingerprint:
+        return None
+
+    # using sha256 only to get a consistent length fingerprint
+    # if it is deemed a performance issue, we can replace it with
+    # with a truncated string as well
+    fingerprint_bytes = str(value_to_fingerprint).encode('utf-8')
+    sha256_hash = hashlib.sha256()
+    sha256_hash.update(fingerprint_bytes)
+    fingerprint = sha256_hash.hexdigest()
+
+    return fingerprint
+
+
+def process_fingerprint(fingerprint: str) -> bool:
+    """
+    does book keeping based on event fingerprint.
+    returns true if the event should be allowed to propagate
+    """
+    # decide whether to send or suppress
+    send_event = False
+    if fingerprint in event_counters:
+        event_count = event_counters[fingerprint][EVENT_COUNT_KEY]
+        suppression_count = event_counters[fingerprint][SUPPRESSION_COUNT_KEY]
```

Comment:
For counters using a [Counter](https://docs.python.org/3/library/collections.html#collections.Counter) is the preferred way as it is thread-safe.

Reply:
sure, will make that change.